tutorials/basic_1_square/README.md

```diff
@@ -0,0 +1,137 @@
+
+Now that we have a better understanding of how the game structure works, lets start to plan our game.
+
+First things first, lets think on what variables and data structure we would need. For that, lets recap what we are building: A simple game where a square keeps bouncing forever from one side of the screen to the other. Thinking on this, we will need:
+
+ - A variable to tell us the speed of the square in logical pixels per second.
+ - A variable to inform which direction the square is moving.
+ - A structure to represent our square, which has a position and dimensions.
+
+With that in mind, check the example below, note the comments for explanations on which code section.
+
+
+```dart
+class MyGame extends Game {
+  // A constant speed, represented in logical pixels per second
+  static const int square_speed = 400;
+
+  // To represent our square we are using the Rect class from dart:ui
+  // which is a handle class to represent this type of data. We will be
+  // seeing other types of data classes in the future, but for this
+  // example, Rect will do fine for us.
+  late Rect squarePos;
+
+  // To reprent our direction, we will be using an int value, where 1 means
+  // going to the right, and -1 going to the left, this may seems like a too much
+  // simple way of representing a direction, and indeed it is, but this will
+  // will work fine for our small example and will make more sense when we implement
+  // the update method
+  int squareDirection = 1;
+
+  // The onLoad method is where all of the game initialization is supposed to go
+  // for this example, you may think that this square could just be initialized on the field
+  // declaration, and you are right, but for ludic reasons and to present the life cycle method
+  // for this example we will be initializing this field here.
+  @override
+  Future<void> onLoad() async {
+    squarePos = Rect.fromLTWH(0, 0, 100, 100);
+  }
+
+  // Update and render ommited
+}
+```
+
+Right, now we have all the data and variables we need to start implementing our game, for the next step, lets just draw our little square on the screen.
+
+```dart
+  @override
+  void render(Canvas canvas) {
+    // Canvas is a class from dart:ui and is reponsible for all the renderizations on Flame
+    //
+    // BasicPalette is a help class from Flame, which provides default, pre builded instances
+    // of Paints that can be used by your game
+    canvas.drawRect(squarePos, BasicPalette.white.paint);
+  }
+}
+```
+
+You may now be seeing a static white square being rendered on the top left corner of your screen, which is not that cool, so, to finish our example, lets add some movement to the game and implement our update method:
+
+```dart
+  @override
+  void update(double dt) {
+    // Here we move our square by applying our iteration delta time (dt) to our
```

Comment:
I think we can double down here on the fact that the update method must update your game state to the next state after an amount of time dt seconds. if you dont consider dt the game will run differently depending on hardware/software conditions

Reply:
Added a new section for this